### PR TITLE
Rattle Me Bones

### PR DESCRIPTION
**Pull request by Shadowmech88**
A spooky vault for the Halloween season.
Adds the Skeleton Den, a new vault in which the objective is to not rattle the skeletons' bones.
skeleton_den
Bumping into any of the skeleton models will cause all the skeleton models in that room to become hostile to you. Some of the skeleton models move around the room on predetermined paths, as seen here:
https://puu.sh/y3sNg/a3baaed5ca.webm
(The stuttering at the end is a video error, not in-game.)
In a couple of the rooms there are skeleton mobs which will attack you by default, requiring you to fight them while still avoiding the skeleton models.

This PR also adds new abstract objects called movers, along with subtypes transporters, redirectors, and speed adjuster. While mapping, a transporter can be placed on the same tile as an object or mob in order to cause it to continuously move in the transporter's `dir` when the game begins. If something with a transporter crosses over a redirector, the transporter's `dir` will change to the redirector's `dir`. If something with a transporter crosses over a speed adjuster, the transporter's `step_delay` will change to the speed adjuster's `step_delay`. This allows mappers to create moving hazards that follow simple or complex courses without ever having to touch any code.

:cl:
 * rscadd: Added a new vault, the skeleton den. Don't rattle their bones!
 * rscadd: Added a new system by which mappers may create moving hazards that follow simple or complex courses without needing to touch any code.

### CODE REVIEW

**Comment by Shadowmech88** [comment]
A couple things I need input on:
* First, would it be *more* or *less* fun if the maze full of things you shouldn't touch, some of which move, was pitch-black and you had to provide your own light source?
Currently the vault is fully-bright inside, but I feel like that does reduce some of the challenge and atmosphere.

* Second, I have no idea what rewards to put in the loot room at the end.

**Comment by jknpj** [comment]
>Second, I have no idea what rewards to put in the loot room at the end.

fifth dissolve injectors ACK ACK!

**Comment by SonixApache** [comment]
A gold medal, special gloves (swat/combat/centcomm) and a money bag full of golden coins. And one of those energy cutlasses.
Do we have non-vox skeleton costumes?

**Comment by Exxion** [comment]
All walls should be at least two tiles thick to prevent disposal memes

**Comment by Shadowmech88** [comment]
I'm just going to make disposal pipe schematics fail when used on super-reinforced walls. I'm tired of people piping through impenetrable barriers.

**Comment by Exxion** [comment]
That works too.

**Comment by FudgePucker** [comment]
Wait I thought you could only make disposal pipes on plating, how are people putting them ON walls? I thought the problem was with it putting you on the wall from the next turf over?

**Comment by Exxion** [comment]
Fuck dude I don't know

File: code/modules/maps/movers.dm [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+	var/step_delay = 1
+
+/obj/abstract/mover/transporter/Destroy()
+	stop_moving()
+	host = null
+	processing_objects.Remove(src)
+	..()
+
+/obj/abstract/mover/transporter/process()
+	if(host)
```
Comment (PJB3005) [inline code/modules/maps/movers.dm]: Merge these two if statements into 1.

**Review by PJB3005 — CHANGES_REQUESTED** [review]

**Comment by Shadowmech88** [comment]
Are they being put onto the next tile by a delivery chute, or by an exposed pipe launching them out?

**Comment by FudgePucker** [comment]
Fun traitor idea: make a SME below cargo in the hidden room next to the mining supplies, reroute disposals to shoot into shard. It produces power for the station by recycling the trash. What do you mean people use disposals as a shortcut across the station?

**Comment by Blithering** [comment]
Do the skeleton mobs rattle the bones when they bump into them? that might make the maze impossible.
Additionally, Couldn't someone bypass the whole maze by unwrenching that disposals exit thing and adding a second entrance to make it two way?

**Comment by Shadowmech88** [comment]
Skeletons are not rattled by being bumped by other skeletons.
As for the disposals, can a person exit a disposal pipe through an inlet?

**Review by PJB3005 — APPROVED** [review]

**Comment by FudgePucker** [comment]
Maybe block it off with shutters like the HoP line ones. Meaning even if you reverse though the disposals you will still be blocked off.

**Comment by unid15** [comment]
Amazing!

**Comment by Shadowmech88** [comment]
Added [Do Not Merge] because it's still WIP.

**Comment by Exxion** [comment]
```
/turf/simulated/wall/invulnerable/Entered(atom/mover)
    qdel(mover)
```

**Comment by Shadowmech88** [comment]
Nobody has offered their thoughts on the lighting situation yet.

**Comment by Shadowmech88** [comment]
Also, I tested it and disposal pipes can't be secured while on a wall tile, meaning walls can't be piped through. I also tried riding through a pipe that terminated at a wall and it just dumped me on the pipe's turf, not the wall's turf. I'm not sure what people are talking about when they mention bypassing walls with disposal pipes.

**Comment by Kitfox88** [comment]
Make it pitch black other than your own light sources. Get Tomb of the Giants in this bitch.

**Comment by Shadowmech88** [comment]
* The loot room of the skeleton den now contains a pirate outfit, gold metal, pirate lord's armor, pirate lord's hat, optical thermal eyepatch, two energy cutlasses, five dissolve injectors, three gold coins, and a bag containing three hundred gold coins.
* The skeleton den has been made pitch-black. Bring a light source.

**Comment by Shadowmech88** [comment]
This is now ready, content-wise.

**Pull request merged**